charts/admission-controller/templates/_helpers.tpl

```diff
@@ -110,26 +110,47 @@ If release name contains chart name it will be used as a full name.
     {{- end -}}
 {{- end -}}
 
+{{/*
+Validate Sysdig API Endpoint
+*/}}
+{{- define "chart.validateSysdigApiEndpoint" -}}
+    {{- if or (hasPrefix "http://" .Values.sysdig.apiEndpoint) (hasPrefix "https://" .Values.sysdig.apiEndpoint) -}}
+        {{- fail "Sysdig API endpoint should not contain http:// or https:// prefix" -}}
+    {{- end -}}
+{{- end -}}
+
 {{/*
 Determine Secure endpoint based on provided region or .Values.sysdig.apiEndpoint
 */}}
 {{- define "admissionController.apiEndpoint" -}}
-    {{- if (or .Values.sysdig.apiEndpoint (eq .Values.global.sysdig.region "custom"))  -}}
-        {{- required "A valid Sysdig API endpoint (.sysdig.apiEndpoint) is required" .Values.sysdig.apiEndpoint -}}
-    {{- else if hasKey ((include "sysdig.regions" .) | fromYaml) .Values.global.sysdig.region }}
-        {{- include "sysdig.secureApiEndpoint" . }}
+    {{ include "chart.validateSysdigApiEndpoint" . }}
```

Comment:
Shouldn't this `include` be inside the first if?

Reply:
This is being validated for the entire define because, in the else scenario, it also should be validated, so instead of validating inside both ifs I'm validating before them.